packages/amplify_core/lib/src/types/storage/remove_options.dart

```diff
@@ -1,14 +1,48 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+import 'package:aws_common/aws_common.dart';
+
 import 'base/storage_operation_options.dart';
 
 /// {@template amplify_core.storage.remove_options}
-/// Configurable options to initiate a [StorageRemoveRequest].
+/// Configurable options for `Amplify.Storage.remove`.
 /// {@endtemplate}
-class StorageRemoveOptions extends StorageOperationOptions {
+class StorageRemoveOptions extends StorageOperationOptions
+    with
+        AWSEquatable<StorageRemoveOptions>,
+        AWSSerializable<Map<String, Object?>>,
+        AWSDebuggable {
   /// {@macro amplify_core.storage.remove_options}
   const StorageRemoveOptions({
     required super.accessLevel,
+    this.pluginOptions,
   });
+
+  /// {@macro amplify_core.storage.remove_plugin_options}
+  final StorageRemovePluginOptions? pluginOptions;
+
+  @override
+  List<Object?> get props => [accessLevel, pluginOptions];
+
+  @override
+  String get runtimeTypeName => 'StorageRemoveOptions';
+
+  @override
+  Map<String, Object?> toJson() => {
+        'accessLevel': accessLevel,
```

Comment:
I don't think enums can be passed to `jsonEncode`

```suggestion
        'accessLevel': accessLevel.name,
```